agent/xdsv2/cluster_resources.go

```diff
@@ -81,7 +80,7 @@ func (pr *ProxyResources) makeClusters(name string) ([]proto.Message, error) {
 	return clusters, nil
 }
 
-func (pr *ProxyResources) makeEnvoyCluster(name string, protocol string, eg *pbproxystate.EndpointGroup) (*envoy_cluster_v3.Cluster, error) {
+func (pr *ProxyResources) makeEnvoyCluster(name string, protocol pbproxystate.Protocol, eg *pbproxystate.EndpointGroup) (*envoy_cluster_v3.Cluster, error) {
```

Comment:
this protocol prevents importing pbcatalog code into xds resource code and keeps the IR as the intermediate seam that we have to keep strong ties between our data model and xds.